tests/integration/test_configuration.py

```diff
@@ -0,0 +1,231 @@
+                  "{{ RPC_RELEASE }}",
+                  "{{ JOB_NAME }}",
+                  "{{ MOLECULE_TEST_REPO }}",
+                  "{{ zz_testcase_class }}"
+                ]
+              }
+            }"""
+
+    with open(filename, 'w') as f:
+        f.write(config_json)
+
+    return filename
+
+
+@pytest.fixture(scope='session')
+def no_module_hierarchy_config(tmpdir_factory):
+    """A configuration missing the test_cycle config
+    """
+
+    filename = tmpdir_factory.mktemp('data').join('config_file.json').strpath
+    config_json = \
+        """{
+              "zigzag": {
+                "test_cycle": "{{ ZZ_INTEGRATION_TEST_CYCLE }}",
+                "project_id": "{{ ZZ_INTEGRATION_PROJECT_ID }}"
+              }
+            }"""
+
+    with open(filename, 'w') as f:
+        f.write(config_json)
+
+    return filename
+
+
+@pytest.fixture(scope='session')
+def no_project_id_config(tmpdir_factory):
+    """A configuration missing the test_cycle config
+    """
+
+    filename = tmpdir_factory.mktemp('data').join('config_file.json').strpath
+    config_json = \
+        """{
+              "zigzag": {
+                "test_cycle": "{{ ZZ_INTEGRATION_TEST_CYCLE }}",
+                "module_hierarchy": [
+                  "{{ RPC_PRODUCT_RELEASE }}",
+                  "{{ RPC_RELEASE }}",
+                  "{{ JOB_NAME }}",
+                  "{{ MOLECULE_TEST_REPO }}",
+                  "{{ zz_testcase_class }}"
+                ]
+              }
+            }"""
+
+    with open(filename, 'w') as f:
+        f.write(config_json)
+
+    return filename
+
+
+@pytest.fixture(scope='session')
+def required_configs(tmpdir_factory):
+    """A configuration with required configs
+    used to test when properties evaluate to empty things
+    """
+
+    filename = tmpdir_factory.mktemp('data').join('config_file.json').strpath
+    config_json = \
+        """{
+              "zigzag": {
+                "test_cycle": "{{ ZZ_INTEGRATION_TEST_CYCLE }}",
+                "project_id": "{{ ZZ_INTEGRATION_PROJECT_ID }}",
+                "module_hierarchy": [
+                  "{{ ZZ_MODULE_HIERARCHY_ONE }}",
+                  "{{ ZZ_MODULE_HIERARCHY_TWO }}"
+                ]
+              }
+            }"""
+
+    with open(filename, 'w') as f:
+        f.write(config_json)
+
+    return filename
+
+
+# ======================================================================================================================
+# Test Suites
+# ======================================================================================================================
+class TestConfig(object):
+    # noinspection PyUnresolvedReferences
+    def test_publish_single_passing_test_with_custom_mod_hierarchy(self, single_passing_test_with_custom_mod_hierarchy):
+        """Verify ZigZag can publish results from the "asc" CI environment with one passing test in the JUnitXML
+        file.
```

Comment:
Description is inaccurate.

Reply:
addressing in fb1987cbd8f4107ef7aba96c807d6cadfa7074dd